File: app/src/at-nhatnguyen/java/asiantech/internship/summer/javacore/Bai5.java
package asiantech.internship.summer.javacore;

import java.util.Scanner;

public class Bai5 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.print("Nhập n = ");
        int n = scan.nextInt();
        System.out.printf("Tất cả các số nguyên tố nhỏ hơn %d là: \n",n);
        if (n >= 2) {
            System.out.print(2);
        }
        for (int i = 3; i < n; i+=2) {
            if (PrimeNumber(i)) {
                System.out.print(" " + i);
            }
        }
    }
    private  static boolean PrimeNumber(int n) {
        if (n < 2) {
            return false;
        }
        for (int i = 2; i <= n-1; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai5.kt
package asiantech.internship.winter.javacore

import java.util.*
import kotlin.math.sqrt

/*
Bài 5. Viết chương trình liệt kê tất cả các số nguyên tố nhỏ hơn n cho trước.
 */
object Bai5 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter n: ")
        val n = scanner.nextInt()
        for (i in 2..n) {
            if (isPrimeNumber(i))
                print("$i ")
        }
    }

    fun isPrimeNumber(n: Int): Boolean {
        if (n < 2) return false
        for (i in 2..sqrt(n.toDouble()).toInt()) {
            if (n % i == 0) return false
        }
        return true
    }
}

